Tests/SwiftDocCTests/Semantics/MetadataSynonymTests.swift

```diff
@@ -0,0 +1,126 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+import Markdown
+
+@testable import SwiftDocC
+
+class MetadataSynonymTests: XCTestCase {
+    /// Basic validity test for giving several directives.
+    func testMultipleSynonymDirectives() throws {
+        let (problems, directive) = try parseDirective(Metadata.self, in: "TestBundle") {
+            """
+            @Metadata {
+                @Synonym(language: objc) {
+                    ``MyKit/MyClass``
+                }
+                @Synonym(language: swift) {
+                    <doc://org.swift.documentation/MyClass/property>
+                }
+            }
+            """
+        }
+        
+        let metadata = try XCTUnwrap(directive)
+        XCTAssertEqual(problems, [])
+        
+        let contents = metadata.synonyms
+            .compactMap { $0.content.first }
+            .map { $0.debugDescription() }
+        XCTAssertEqual(contents, [
+            """
+            └─ Paragraph
+            └─ SymbolLink destination: MyKit/MyClass
+            """,
+            """
+            └─ Paragraph
+            └─ Link destination: "doc://org.swift.documentation/MyClass/property"
+               └─ Text "doc://org.swift.documentation/MyClass/property"
+            """
+        ], "Received: \n\(contents.joined(separator: "\n"))")
+        
+        let links = metadata.synonyms.map(\.link)
+        XCTAssertEqual(links.count, 2)
+        
+        let symbolLink = try XCTUnwrap(links.first as? SymbolLink)
+        XCTAssertEqual(symbolLink.destination, "MyKit/MyClass")
+        
+        let link = try XCTUnwrap(links.last as? Link)
+        XCTAssertEqual(link.destination, "doc://org.swift.documentation/MyClass/property")
+    }
+    
+    func testDuplicateSynonymDirectives() throws {
+        let (problems, _) = try parseDirective(Metadata.self, in: "TestBundle") {
+            """
+            @Metadata {
+                @Synonym(language: objc) {
+                    ``MyClass/property``
+                }
+                @Synonym(language: objc) {
+                    <doc://org.swift.documentation/MyClass/property>
+                }
+            }
+            """
+        }
+        
+        XCTAssertEqual(problems, [
+            "2: warning – org.swift.docc.Synonym.DuplicateLanguage",
+            "5: warning – org.swift.docc.Synonym.DuplicateLanguage"
+        ])
+    }
+    
+    func testObjCAliasesInSynonymDirective() throws {
+        for alias in SourceLanguage.objectiveC.idAliases {
+            let (problems, directive) = try parseDirective(Metadata.self, in: "TestBundle") {
```

Comment:
minor: It would be good if we didn't create a full bundle in a loop just to test parsing a directive. AFAICT this test doesn't validate that the link resolved so it would work the same without the bundle. 

Also, please don't use "TestBundle" in any new test. It doesn't represent realistic inputs.